proposal.rst

```diff
@@ -237,7 +247,51 @@ Project Timeline
 .. any university classes or exams, vacations, etc), make sure you include them
 .. in your project timeline.
 
-FILLME
+* Pre-acceptance Period from Present to May 5th
+	This one month period can be utilized to work on issues related to omega so that I can get familiar to the codebase and also to gain some familiarity to Linux system programming.
+	
+* Community Bonding Period from May 6th to May 26th
+	Understand the usage of the shared libraries to be used to replace external filter programs by reading the documentation and discussing it with mentors and continue working on issues related to omega.
+
+					 ----- Coding officially begins -----
+
+* Week 1 from May 27th to June 2nd
+	Testing Omega before adding any library to the project and check how it runs with the verbose mode to get the grasp of the sequence of implementation of current source code. Continue search and understanding the usage of libraries. Discuss the outcomes with mentors.
+
+* Week 2 and 3 from June 3rd to June 16th
+	Implementing the libarchive library for reading zip file formats. Testing and documentation along. (This phase may take a longer time than expected as this is the beginning of coding)
+
+* Week 4 from June 17th to June 23rd
+	Implementing the libpoppler library for reading zip file formats. Testing and documentation along. Getting libarchive and libpoppler to work before Phase I evaluation will be my main goal.
+
+					 ----- Phase 1 Evaluation -----
+
+* Week 5 and 6 from June 24th to July 7th
+	Implementing the DjVuLibre library for reading DjVu format. Testing and documentation along.
+
+* Week 7 from July 8th to July 14th
+	Although the libarchive would be able to read the AbiWord files, the extent of AbiWord is large. Hence, I would like to implement the library libabw in case the library crashes. This week can be devoted to it. Testing and documentation all along.
+
+* Week 8 from July 15th-July 21st
```

Comment:
It feels like everything from here out are as-yet undefined stretch goals. One thing I'd suggest is to put some padding _earlier_ in the project, to ensure that there's time to get early work merged. From experience, the first merge of a GSoC project is often a bit fiddly and may take longer than anticipated, so having time before the first evaluation (alongside other work, but pacing it slower because of the need to address comments &c) is a good idea.

Reply:
I had thought about it too but I thought that implementing both libarchive and libpoppler before the phase 1 evaluation would be good so I planned accordingly. Now that you suggest, I shall change it.